biopet-core/src/test/scala/nl/lumc/sasc/biopet/core/report/MultisampleReportBuilderTest.scala

```diff
@@ -41,10 +45,25 @@ class MultisampleReportBuilderTest extends TestNGSuite with Matchers {
 
       def libraryPage(sampleId: Int, libraryId: Int, args: Map[String, Any]) = ReportPage(Nil, Nil, Map())
     }
+    val dbFile = File.createTempFile("summary.", ".db")
+    dbFile.deleteOnExit()
+    val db = SummaryDb.openSqliteSummary(dbFile)
+    db.createTables()
+
+    val sample = Some("sampleName")
+    val lib = Some("libName")
+
+    sample.foreach { sampleName =>
+      val sampleId = Await.result(db.createSample(sampleName, 0), Duration.Inf)
```

Comment:
This is a test, I realize, but `createSample` is being called in a `.foreach` loop here and immediately used for `Await`. This means the `Futures` are actually launched one after another, so you don't get the benefit of async here.